TechAndCheck/zenodotus
# Let new users complete their account setup

This PR adds the final piece to #271: letting newly-approved users receive an email with their setup instructions, with a link to choose a password and come on in. As with the previous PRs in this chain, testing it is a little bit clunky until we complete #273.

🚨 Note that this PR now includes some new database seeds, so you may want to re-seed. (You don't have to, though; read the prerequisites block. However, future PRs will likely assume you have re-seeded and will make use of these seeds.)

**Testing prerequisite:**

To test this, you need a **newly-created applicant** that has **confirmed their email address** and been **approved by an admin**, and then been **turned into a user**. You can either go through that process manually:

1. Go to http://localhost:3000/apply and apply for a new account (with an email address you control)
2. Check your email, click the link, confirm their email address
3. Switch to the Rails console: `rails c`
4. Find that applicant (`a = Applicant.find(…)`) and approve them: `a.approve`
5. Create a user from that now-approved applicant: `u = User.create_from_applicant(a)`

Or you can re-seed your database to pick up some new seeds, which shaves some steps off. You still have to start with an applicant, so you can update it to have an email address you can receive email at before creating the user:

```ruby
a = Applicant.find_by(email: "approved-applicant@example.com")
a.update!(email: "some-email@you-can-access")
u = User.create_from_applicant(a)
```

Whichever path you choose, you can now continue on below.

**Testing:**
1. Start Rails and Sidekiq
2. Send setup instructions to that user: `u.send_setup_instructions`
3. Check your email, click the link, and choose your password
4. ✅ You should be logged in!

**Testing failing cases:**
- Try bad passwords
- Try to setup an account a second time

Closes #271
